KK-002 Changed Footer

eleniaslanidou89 opened the pull request:
- changed text
- new styling

eleniaslanidou89 committed:
changed Footer Component

changed text and styling

kgkoutio reviewed (commented):
Hey @eleniaslanidou89 ! I added a few comments in the PR. Feel free to ignore if they make no sense to you. One think I noticed when I run the project is that the old footer is used instead. Maybe I did something wrong, but this is what I get:
<img width="1271" alt="Screen Shot 2021-10-30 at 13 11 46" src="https://user-images.githubusercontent.com/93218410/139530606-7888e547-808d-4b31-8f76-bc1a6901585e.png">

eleniaslanidou89 commented on src/components/Footer.vue:
```diff
@@ -1,61 +1,107 @@
+}
+
+.footer-logo_text-bold {
+  text-transform: uppercase;
```
Yes, it will be bold so I will add it.

eleniaslanidou89 commented on src/components/Footer.vue:
```diff
@@ -1,61 +1,107 @@
+
+.footer-nav_link:hover,
+.footer-nav_link:focus {
+  color: #fff;
```
I believe that both rgb and hex are commonly used. But we will follow your suggest, naming the colors :)

kgkoutio replied:
Yes you are right, they are both valid and commonly used. But are they used both on the same project? I searched for best practises and I found this: https://vuejs.org/v2/style-guide/
This mentions as a priority C that when there are equally good options, consistency should be considered. That is actually my point here. I would recommend to choose what style you like the most and use that in the project.

eleniaslanidou89 commented on src/components/Footer.vue:
```diff
@@ -1,61 +1,107 @@
-        </li>
-      </ul>
+  <footer class="footer">
+    <div class="footer_container">
```
I am using _ as a kind of secondary name class. But I will change the name of them to more descriptive names when I will create the new footer.

kgkoutio replied:
What is a secondary name class?

eleniaslanidou89 replied:
By secondary name of class I mean a second name for the class in order to give more details about it. For example footer_nav_list, the list is something like a secondary part of the name. Let me know if this isn't clear. Maybe I should change the way that I am naming the classes, if this makes confusion. 😏

kgkoutio replied:
Hmmm yes it is not very intuitive. Naming is one of the hardest things. I think every class should be named after what it is supposed to do, in a way that is self-explanatory what this class does. I am very confused with the - and _. Intuitively it feels wrong to be inconsistent in the naming of variables and classes. Of course, there might be a web development naming pattern that I am not aware of :) @kkarayannis any idea?

kkarayannis replied:
Just to see what other people do I looked at some open source projects listed [here](https://www.devaradise.com/vue-example-projects). Now, it could be that I didn’t look hard enough, but not only did I not find any mixed `-` and `_` uses, but I didn’t see any `_` used at all. Everyone seems to just be using `-` everywhere. Other than that, the names are ok. `footer-nav-list` and `footer-nav-item` are descriptive enough IMO.

eleniaslanidou89 replied:
I have found that there is a css name convention called BEM (block, element, modifier). You can look it up, there are specific rules for it :) [->](http://getbem.com/introduction/)

kgkoutio replied:
Ah thank you @eleniaslanidou89 ! This article explains very well the BEM architecture. If you like it we can use it in the whole project. In the screenshot above the naming is very similar to the one in the article, but the naming in the `Footer.vue` seems a bit different. Should we follow the convention stated in the article in the whole project? What do you think?

eleniaslanidou89 replied:
Yes, we will follow it in the whole project :) I will change the names in the Footer.vue too.

eleniaslanidou89 committed:
deleted useless icons

eleniaslanidou89 committed:
added new images

eleniaslanidou89 committed:
created new containers and added images

eleniaslanidou89 committed:
added link and jQuery for Bootstrap

eleniaslanidou89 committed:
changes on footer's width

eleniaslanidou89 committed:
🌻 Added google font and new image(flowers)

eleniaslanidou89 committed:
added colors

eleniaslanidou89 committed:
Defined colors

eleniaslanidou89 committed:
Deleted color on hover

eleniaslanidou89 committed:
Added padding

eleniaslanidou89 committed:
Changed responsive design

eleniaslanidou89 commented:
![image](https://user-images.githubusercontent.com/57090719/141965848-1d8ab4da-97de-4591-a94a-6f34419aa523.png)

Footer Section
- Added new images and deleted the useless.
- Changed styling.
- Added responsive design.
- Improved structure(class names, defined color names).
- Added Bootstrap library.

eleniaslanidou89 committed:
Changed background color

eleniaslanidou89 commented:
![image](https://user-images.githubusercontent.com/57090719/141992731-7be20fe7-cd8c-4644-8567-7e66ad4692cc.png)

Changed background color

eleniaslanidou89 committed:
Added padding and changed background color

eleniaslanidou89 committed:
Merge branch 'main' into KK-002
